tests/rest/devices.spec.js

```diff
@@ -0,0 +1,211 @@
+const chai = require('chai');
+const chai_http = require('chai-http');
+const should = chai.should();
+var server = require('./server-if');
+
+chai.use(chai_http);
+
+describe('Devices interface', () => {
+
+  before(() => {
+    server.start();
+  });
+
+  after(() => {
+  });
+
+  describe('POST /devices', function() {
+    it('should return 201', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":"cHNrMw==","psk_id":"cHNraWQz","uuid":"GHI"}')
+        .end( (err, res) => {
+          should.not.exist(err);
+          res.should.have.status(201);
+          done();
+        });
+    });
+
+    it('with empty payload should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with an array instead of an object should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('[{"psk":"cHNrMQ==","psk_id":"cHNraWQx","uuid":"ABC"}, {"psk":"cHNrMg==","psk_id":"cHNraWQy","uuid":"DEF"}]')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with missing key in payload should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk_id":"cHNraWQx","uuid":"ABC"}')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with invalid base64 string should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":"invalid-base64-string","psk_id":"cHNraWQx","uuid":"ABC"}')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with invalid value at key should return 400', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":true,"psk_id":"cHNraWQx","uuid":"ABC"}')
+        .end( (err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+
+    it('with additional invalid key should return 201', (done) => {
+      chai.request(server)
+        .post('/devices')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":"cHNrNA==","psk_id":"cHNraWQ0","uuid":"JKL", "invalid-key":"invalid-value"}')
+        .end( (err, res) => {
+          should.not.exist(err);
+          res.should.have.status(201);
```

Comment:
```javascript
res.body.should.be.a('object');
res.body.should.have.property('uuid'); // redundant test
res.body.should.have.property('psk_id'); // redundant test
res.body.should.not.have.property('invalid-key'); // this can also be uuid
```